src/components/RegionFilter.jsx

```diff
@@ -0,0 +1,44 @@
+import 'react';
+import {useState} from "react";
+
+/**
+ * RegionFilter Component
+ * Displays a dropdown menu for filtering by region.
+ *
+ * Props:
+ * - action: Function to handle the selection of a region.
+ *
+ * Functionality:
+ * - Toggles the dropdown menu open and closed.
+ * - Calls the provided `action` function when a region is clicked.
+ */
+
+const RegionFilter = ({action}) => {
+    const [isOpen, setIsOpen] = useState(false);
+
+
+    const toggleDropdown = () => {
+        setIsOpen(!isOpen);
+    }
+
+    const regions = ["All", "Africa", "Americas", "Asia", "Europe", "Oceania"];
+
+
+    return (
+        <div className={`dropdown-wrapper ${isOpen && "open"}`}>
+            <div className="dropdown-header flex flex-jc-sb flex-ai-c" onClick={toggleDropdown}>
+                <span>Filter by Region</span>
+                <i className="fa-solid fa-chevron-down icon"></i>
+            </div>
+            <div className="dropdown-body">
+                <ul onClick={(e) => action(e.target.getAttribute("data-region"))}>
+                    {regions.map((region) => (
+                        <li key={region} data-region={region}>{region}</li>
+                    ))}
```

Comment:
Is it possible for 2 countries to be in the same region? 

If yes then think about different key for the  li ( we don`t want to have 2 of the same key)